python/cog/server/worker.py

```diff
@@ -179,6 +306,8 @@ def run(self) -> None:
         ws_stdout.wrap()
         ws_stderr.wrap()
 
+        # using a thread for this can potentially cause a deadlock
+        # however, if we made this async, we might interfere with a user's event loop
```

Comment:
My earlier idea was to run `non-async def predict` code in a thread without an event loop. That way we don't break any legacy event loops. That won't make it easier to kill though.